app/services/processing_service.py

```
@@ -450,44 +523,282 @@ async def prepare_repository(self, repo_name) -> Tuple[Path, str]:
+        except Exception as e:
+            logger.warning(f"Failed to resolve dependencies for {repo_name}: {e}")
+
+        return cloned_paths
+
     def clone_and_process_repository(
-        self, repo_url: str, repo_path: str, branch: str = "main"
+            self,
+            repo_url: str,
+            repo_path: str,
+            branch: str = "main",
+            commit_hash: str = None
```

Ruff (0.14.4)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)